tests/Doctrine/Tests/ORM/Query/SelectSqlGenerationTest.php

```diff
@@ -84,6 +84,23 @@ public function assertInvalidSqlGeneration($dqlToBeTested, $expectedException, a
         $this->fail($sql);
     }
 
+    public function testDDC3697WithPartialLoad()
```

Comment:
I would add (to the method name) something that indicates that `WHERE` and `WITH` both work